inference-engine/ie_bridges/java/cpp/blob.cpp

```diff
@@ -0,0 +1,143 @@
+#include <inference_engine.hpp>
+#include "openvino_java.hpp"
+#include "jni_common.hpp"
+
+using namespace InferenceEngine;
+
+JNIEXPORT jfloatArray JNICALL Java_org_intel_openvino_Blob_asFloat(JNIEnv *env, jobject obj, jlong addr)
+{
+    static const char method_name[] = "asFloat";
+
+    try
+    {
+        Blob::Ptr *output = reinterpret_cast<Blob::Ptr *>(addr);
+        const float *buffer = as<MemoryBlob>(*output)->rmap().as<const float *>();
+
+        jfloatArray result;
+        int size = (*output)->size();
+        result = env->NewFloatArray(size);
+
+        env->SetFloatArrayRegion(result, 0, size, buffer);
+
+        return result;
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+
+    return 0;
+}
+
+JNIEXPORT jbyteArray JNICALL Java_org_intel_openvino_Blob_asByte(JNIEnv *env, jobject obj, jlong addr)
+{
+    static const char method_name[] = "asByte";
+
+    try
+    {
+        Blob::Ptr *output = reinterpret_cast<Blob::Ptr *>(addr);
+        const uint8_t *buffer = as<MemoryBlob>(*output)->rmap().as<const uint8_t *>();
+
+        jbyteArray result;
+        int size = (*output)->size();
+        result = env->NewByteArray(size);
+
+        env->SetByteArrayRegion(result, 0, size, reinterpret_cast<const jbyte*>(buffer));
+
+        return result;
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+
+    return 0;
+}
+
+JNIEXPORT jlong JNICALL Java_org_intel_openvino_Blob_GetTensorDesc(JNIEnv *env, jobject obj, jlong addr)
+{
+    static const char method_name[] = "GetTensorDesc";
+
+    try
+    {
+        Blob::Ptr *output = reinterpret_cast<Blob::Ptr *>(addr);
+        InferenceEngine::TensorDesc *tDesc = new TensorDesc((*output)->getTensorDesc());
+
+        return (jlong)tDesc;
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+
+    return 0;
+}
+
+JNIEXPORT jlong JNICALL Java_org_intel_openvino_Blob_BlobByte(JNIEnv *env, jobject obj, jlong tensorDescAddr, jbyteArray data)
+{
+    static const char method_name[] = "BlobByte";
+
+    try
+    {
+        InferenceEngine::TensorDesc *tDesc = (InferenceEngine::TensorDesc *)tensorDescAddr;
+
+        Blob::Ptr *blob = new Blob::Ptr();
+        *blob = InferenceEngine::make_shared_blob<uint8_t>( (*tDesc),
```

Comment:
Why do you use the name "BlobByte"?

Why don't you call the allocate method for the blob?

Reply:
@ilyachur, do you mean `blob->allocate`, right?

`BlobByte` is just internal name, it's not available outside.

Reply:
Yes, I understood that it is an internal name. No question to it.

Yes, I mean `blob->allocate()`. As I can see this required method doesn't use in [examples](https://github.com/openvinotoolkit/openvino/pull/506/files#diff-e839a7de4680f3e5382a0dbe94554194R38).

We need to call it separately as in C++ or call it here.

Reply:
We will take a look, thanks! This function implements a wrapper for allocated memory. After we finish with asynchronous sample we will work on Blobs hierarchy as well (see the tasks in PR's description).